CMakeLists.txt.in

```diff
@@ -0,0 +1,215 @@
+cmake_minimum_required(VERSION 3.5)
+project(build-deps NONE)
+
+cmake_host_system_information(
+  RESULT build_concurrency_factor
+  QUERY NUMBER_OF_LOGICAL_CORES)
+set(make_command make -j ${build_concurrency_factor})
+
+include(ExternalProject)
+
+ExternalProject_Add(boost_ext
+  # The 1.67.0 release has a bug in Boost Lockfree around a missing header.
+  URL https://dl.bintray.com/boostorg/release/1.64.0/source/boost_1_64_0.tar.gz
+  URL_MD5 319c6ffbbeccc366f14bb68767a6db79
+  PATCH_COMMAND
+    ./bootstrap.sh
+    --prefix=@SMF_DEPS_INSTALL_DIR@
+    --with-libraries=atomic,chrono,date_time,filesystem,program_options,system,test,thread
+  CONFIGURE_COMMAND ""
+  BUILD_COMMAND ""
+  INSTALL_COMMAND
+    ${CMAKE_COMMAND} -E chdir <SOURCE_DIR>
+    ./b2
+    -j ${build_concurrency_factor}
+    --layout=system
+    --build-dir=<BINARY_DIR>
+    install
+    variant=debug
+    link=shared
+    threading=multi
+    hardcode-dll-paths=true
+    dll-path=@SMF_DEPS_INSTALL_DIR@/lib)
+
+ExternalProject_Add(fmt_ext
+  GIT_REPOSITORY https://github.com/fmtlib/fmt.git
+  GIT_TAG        9e554999ce02cf86fcdfe74fe740c4fe3f5a56d5
+  INSTALL_DIR    @SMF_DEPS_INSTALL_DIR@
+  CMAKE_ARGS
+    -DCMAKE_BUILD_TYPE=Release
+    -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
+    -DFMT_INSTALL=ON
+    -DFMT_DOC=OFF
+    -DFMT_TEST=OFF
+  BUILD_COMMAND ${make_command})
+
+ExternalProject_Add(protobuf_ext
+  URL https://github.com/protocolbuffers/protobuf/releases/download/v3.3.0/protobuf-cpp-3.3.0.tar.gz
+  URL_MD5 73c28d3044e89782bdc8d9fdcfbb5792
+  INSTALL_DIR @SMF_DEPS_INSTALL_DIR@
+  CONFIGURE_COMMAND <SOURCE_DIR>/configure --prefix=<INSTALL_DIR> --srcdir=<SOURCE_DIR>
+  BUILD_COMMAND ${make_command})
+
+# FIXME: this setup seems to cause dpdk to rebuild each time
+set(dpdk_quadruple ${CMAKE_SYSTEM_PROCESSOR}-native-linuxapp-gcc)
+
+set (dpdk_args
+  EXTRA_CFLAGS=-Wno-error
+  O=<BINARY_DIR>
+  DESTDIR=@SMF_DEPS_INSTALL_DIR@
+  T=${dpdk_quadruple})
+
+ExternalProject_Add(dpdk_ext
+  GIT_REPOSITORY https://github.com/scylladb/dpdk.git
+  GIT_TAG a1774652fbbb1fe7c0ff392d5e66de60a0154df6
+  CONFIGURE_COMMAND
+    COMMAND
+      ${CMAKE_COMMAND} -E chdir <SOURCE_DIR>
+      make ${dpdk_args} config
+    COMMAND
+      ${CMAKE_COMMAND}
+      -DSeastar_DPDK_CONFIG_FILE_IN=<BINARY_DIR>/.config
+      -DSeastar_DPDK_CONFIG_FILE_CHANGES=@CMAKE_CURRENT_SOURCE_DIR@/dpdk_config
+      -DSeastar_DPDK_CONFIG_FILE_OUT=<BINARY_DIR>/${dpdk_quadruple}/.config
+      -P @CMAKE_CURRENT_SOURCE_DIR@/cmake/dpdk_configure.cmake
+  BUILD_COMMAND ""
+  INSTALL_COMMAND
+   ${CMAKE_COMMAND} -E chdir <SOURCE_DIR>
+   ${make_command} ${dpdk_args} install)
+
+ExternalProject_Add(c-ares_ext
+  URL https://c-ares.haxx.se/download/c-ares-1.13.0.tar.gz
+  URL_MD5 d2e010b43537794d8bedfb562ae6bba2
+  INSTALL_DIR @SMF_DEPS_INSTALL_DIR@
+  CONFIGURE_COMMAND <SOURCE_DIR>/configure --prefix=<INSTALL_DIR> --srcdir=<SOURCE_DIR>
+  BUILD_COMMAND ""
```

Comment:
btw; i read somewhere that there is a diff between ` "" vs <DISABLE>`

Reply:
I think this is a cooking.sh thing https://github.com/hakuch/CMakeCooking/blob/master/cooking.sh#L634

Reply:
Let me know if you recall where you saw that. I wasn't able to find any info on it.

Reply:
thanks for finding this ref. You are right. I was wrong.